Buildings/Resources/Documentation/userGuide/source/development.rst

```diff
@@ -170,18 +539,38 @@ The following rules need to be followed, in addition to the guidelines described
    Do not use ``Dialog(tab=STRING))``, unless the parameter is declared with a default value
    and this parameter and its value is of no interest to typical users.
 
-#. Each block must have a ``defaultComponentName`` annotation.
+#. In the code text the instances must be ordered as follows:
+     - If an instance has a type associated with it, then list them in this order: `boolean`, `integer`, `real`.
+     - Instances of lower dimensionality should come first, e.g. scalar values before arrays, 
+       though grouping based on model specific similarities may be prioritized
+     - Instances should be ordered as: parameters, interfaces: first inputs, then outputs, then ports,
+       followed by any blocks/instances
+     - Protected instances are below all the unprotected instances and follow the same instance ordering rules.
+
+#. Each block must have a ``defaultComponentName`` annotation and a ``%name`` at icon layer
 
 #. To aid readability, the formatting of the Modelica source code file must be consistent with other
    implemented blocks, e.g., use two spaces for indentation (no tabulators),
-   assign each parameter value on a new line, list first all parameters, then all inputs, all
-   outputs, and finally all instances of other blocks.
+   assign each parameter value on a new line. It is recommended to add an empty line between instances.
    See for example
    `G36_PR1.AHUs.SingleZone.VAV.SetPoints.ExhaustDamper <https://github.com/lbl-srg/modelica-buildings/blob/94d5919dbe1b2f2e317e7b69800f3b3ad07be930/Buildings/Controls/OBC/ASHRAE/G36_PR1/AHUs/SingleZone/VAV/SetPoints/ExhaustDamper.mo>`_.
 
 #. For parameters, where generally valid values can be provided, provide them
    as default values.
 
+#. Note that the protected instances are not accessible with the plots used in the validation tests.
+
+#. Add comments to all the instances such as parameters, interfaces, blocks. The comments should be short and concise. The comments 
```

Comment:
There are no interfaces, and "concise" is by definition short.

Reply:
Agreed